casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -606,4 +606,45 @@ object BlockAPI {
       )
     )
   }
+
+  def isFinalized[F[_]: Monad: EngineCell: SafetyOracle: BlockStore: Log](
+      request: IsFinalizedQuery
+  ): Effect[F, IsFinalizedResponse] = {
+    val errorMessage =
+      "Could not check if block is finalized, casper instance was not available yet."
+    EngineCell[F].read >>= (
+      _.withCasper[ApiErr[IsFinalizedResponse]](
+        implicit casper =>
+          for {
+            lastFinalizedBlock <- casper.lastFinalizedBlock
+            lastFinalizedBlockMetadata = BlockMetadata
+              .fromBlock(lastFinalizedBlock, invalid = false)
+            dag                   <- casper.blockDag
+            givenBlockHash        = ProtoUtil.stringToByteString(request.hash)
+            givenBlockMetadataOpt <- dag.lookup(givenBlockHash)
+            result <- givenBlockMetadataOpt match {
+                       case None =>
+                         s"Could not find block with hash ${request.hash}"
+                           .asLeft[IsFinalizedResponse]
+                           .pure[F]
+                       case Some(givenBlockMetadata) =>
+                         DagOperations
+                           .lowestCommonAncestorF[F](
+                             lastFinalizedBlockMetadata,
+                             givenBlockMetadata,
+                             dag
+                           )
+                           .map { lca =>
+                             if (lca == givenBlockMetadata) {
+                               IsFinalizedResponse(isFinalized = true).asRight
+                             } else {
+                               IsFinalizedResponse(isFinalized = false).asRight
+                             }
```

Comment:
This could just be `IsFinalizedResponse(isFinalized = lca == givenBlockMetadata).asRight`